datahub-graphql-core/src/main/resources/search.graphql

```diff
@@ -1303,4 +1315,30 @@ input SortCriterion {
   The order in which we will be sorting
   """
   sortOrder: SortOrder!
+}
+
+"""
+A grouping specification for search results
+"""
+input GroupingSpec {
+
+  groupingCriteria: [GroupingCriterion!]!
+
+}
+
+"""
+A single grouping criterion for grouping search results
+"""
+input GroupingCriterion {
+
+  """
+  The raw entity type that needs to be grouped
+  """
+  rawEntityType: String!
```

Comment:
EntityType -> GraphQL entity type enum 

here is an example:

https://github.com/datahub-project/datahub/blob/master/datahub-graphql-core/src/main/resources/search.graphql#L155

Conversion: 

https://github.com/datahub-project/datahub/blob/master/datahub-graphql-core/src/main/java/com/linkedin/datahub/graphql/resolvers/search/SearchAcrossLineageResolver.java#L58

also, maybe we can call this baseEntityType